### PR TITLE
Bug 1734390: Extend auth regex for prometheus proxy

### PR DESCRIPTION
**Pull request by paulfantom**
Without it telemeter-client has problems connecting to prometheus.

/cc @LiliC @s-urbaniak @stlaz

### CODE REVIEW

**Comment by stlaz** [comment]
/lgtm
/hold
remove hold when the prometheus-proxy logs are error-free

**Comment by paulfantom** [comment]
It didn't work, errors are still present in logs: https://storage.googleapis.com/origin-ci-test/pr-logs/pull/openshift_cluster-monitoring-operator/430/pull-ci-openshift-cluster-monitoring-operator-master-e2e-aws/1231/artifacts/e2e-aws/pods/openshift-monitoring_prometheus-k8s-0_prometheus-proxy.log

**Comment by stlaz** [comment]
Interesting, the same seems to be fixing the problem in my cluster, can you please add `-request-logging` to the `oauth-proxy` flags?
Perhaps we're missing something that only happens at early stages of the proxy run

File: assets/prometheus-k8s/prometheus.yaml [hunk 1]
```diff
@@ -51,7 +51,8 @@ spec:
     - -cookie-secret-file=/etc/proxy/secrets/session_secret
     - -openshift-ca=/etc/pki/tls/cert.pem
     - -openshift-ca=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt
-    - -skip-auth-regex=^/metrics
+    - -skip-auth-regex=^/(metrics|federate)
```
Comment (s-urbaniak) [inline assets/prometheus-k8s/prometheus.yaml]: we do want to protect the `/federate` endpoint.

**Comment by enj** [comment]
`/metrics` should require auth.

**Comment by s-urbaniak** [comment]
skipping auth for `/metrics` is something that predates me unfortunately, maybe @brancz has some more historical context.

**Comment by s-urbaniak** [comment]
@paulfantom i guess this can be closed out, correct? as this was just a logging issue?

**Comment by stlaz** [comment]
/close
Yes please

**Pull request closed without merging**